### PR TITLE
Add foreign keys to databases and fix tests (Solves: GitOps-1461)

### PR DESCRIPTION
**Pull request by samyak-jn**
Hi :hugs:

This pull request is in with regards to the story GitOps-1461, the story deals with adding foreign keys to database schema and fixing tests accordingly!

### CODE REVIEW

File: db-schema.sql [hunk 1]
```diff
@@ -292,6 +242,60 @@ CREATE TABLE ApplicationState (
...
+
+	PRIMARY KEY(kubernetes_resource_type, kubernetes_resource_uid, db_relation_type, db_relation_key)
+);
+
+-- TODO: Add index for
+-- - kubernetes_resource_type VARCHAR(64) NOT NULL,
+-- - kubernetes_resource_uid  VARCHAR(64) NOT NULL,
+-- - db_relation_type  VARCHAR(64) NOT NULL,
+
+CREATE INDEX idx_resource_type ON KubernetesToDBResourceMapping(kubernetes_resource_type);
```
Comment (jgwest) [inline db-schema.sql:295]: Whoops, I should have been more descriptive in my TODO! This should be a single index containing all three fields mentioned in the TODO. This allows us to do quick lookups  of (k8s resource type, k8s resource uid, db_relation_type) -> db_relation_uid, which I have found I am doing a fair bit of in the code at the moment.

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great! I am glad that we didn't need to do a lot of test changes for this :tada: (I guess when I wrote the test code I was careful to define things within proper ownership order.)

Just one tweak to one of the indexes, then good to go.

File: db-schema.sql (hunk 1 above)
Comment (samyak-jn) [reply db-schema.sql]: Makes sense! :hugs:

**Review by samyak-jn — COMMENTED** [review]
Hiya, thank you for reviewing this! I have pulled the main to resolve conflicts as well, and all tests are passing!

**Review by jgwest — APPROVED** [review]
Rebased and approved, thanks @samyak-jn!

**Pull request merged**